Tchap/Extensions/MXRoomSummary.swift

```diff
@@ -72,11 +65,13 @@ enum RetentionConstants {
                         self.others[Constants.roomAccessRuleKey] = rule
                         updated = true
                     }
-                } else if type == RoomService.roomRetentionStateEventType {
+                } else if type == MXEventType.roomRetention.identifier {
                     if let maxLifetime = event.content[RoomService.roomRetentionContentMaxLifetimeKey] as? UInt64 {
                         self.others[Constants.roomRetentionInDaysKey] = Tools.numberOfDaysFromDuration(inMs: maxLifetime)
-                        updated = true
+                    } else {
+                        self.others[Constants.roomRetentionInDaysKey] = RetentionConstants.undefinedRetentionValueInDays
```

Comment:
Are you suggesting that `maxLifetime` could be null ?

Reply:
yes, it is possible that there is no content in the event
in this case maxLifetime is null (undefined) that is why I use undefinedRetentionValueInDays (=uint.max)